actions/class.ItemContent.php

```diff
@@ -24,70 +24,50 @@
 use oat\tao\model\accessControl\data\PermissionException;
 use oat\tao\model\http\ContentDetector;
 use oat\tao\model\media\MediaBrowser;
+use oat\tao\model\media\mediaSource\DirectorySearchQuery;
+use oat\taoItems\model\media\AssetTreeBuilder;
 use oat\taoItems\model\media\ItemMediaResolver;
 
 /**
  * Items Content Controller provide access to the files of an item
  *
  * @author Joel Bout, <joel@taotesting.com>
- * @package taoItems
  */
 class taoItems_actions_ItemContent extends tao_actions_CommonModule
 {
     use OntologyAwareTrait;
 
     /**
-     * Returns a json encoded array describign a directory
-     *
      * @throws common_exception_MissingParameter
-     * @return string
      */
-    public function files()
+    public function files(): void
     {
-        if (!$this->hasRequestParameter('uri')) {
+        $params = $this->getPsrRequest()->getQueryParams();
+
+        if (empty($params['uri'])) {
             throw new common_exception_MissingParameter('uri', __METHOD__);
         }
-        $itemUri = $this->getRequestParameter('uri');
-        $item = $this->getResource($itemUri);
 
-        if (!$this->hasRequestParameter('lang')) {
+        if (empty($params['lang'])) {
             throw new common_exception_MissingParameter('lang', __METHOD__);
         }
-        $itemLang = $this->getRequestParameter('lang');
 
-        //build filters
-        $filters = [];
-        if ($this->hasRequestParameter('filters')) {
-            $filterParameter = $this->getRequestParameter('filters');
-            if (is_array($filterParameter)) {
-                foreach ($filterParameter as $filter) {
-                    if (preg_match('/\/\*/', $filter['mime'])) {
-                        $this->logWarning('Stars mime type are not yet supported, filter "' . $filter['mime'] . '" will fail');
-                    }
-                    $filters[] = $filter['mime'];
-                }
-            } else {
-                if (preg_match('/\/\*/', $filterParameter)) {
-                    $this->logWarning('Stars mime type are not yet supported, filter "' . $filterParameter . '" will fail');
-                }
-                $filters = array_map('trim', explode(',', $filterParameter));
-            }
-        }
-        $depth = $this->hasRequestParameter('depth') ? $this->getRequestParameter('depth') : 1;
+        $itemLang = $params['lang'];
+        $path = $params['path'];
```

Comment:
do we need to check this or it is optional ?

Reply:
Good catch, It's required, but original code has no validation on it; so adding it